### ISSUE TITLE
The Legend of Heroes: Trails of Cold Steel II (748490)

### ISSUE THREAD

**Issue by enimog** [ValveSoftware/Proton#2134]
# Compatibility Report
- Name of the game with compatibility issues: The Legend of Heroes: Trails of Cold Steel II 
- Steam AppID of the game: 748490

## System Information
- GPU: GTX970
- Driver/LLVM version: nvidia 415.25
- Kernel version: 4.19.11-1-ck-skylake
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/enimog/da49f9c15db8e9c159dfd280cd6a575a (This is with older drivers but still same hardware info)
- Proton version: 3.16-6 beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
The following issues apply for both The Legend of Heroes: Trails of Cold Steel and The Legend of Heroes: Trails of Cold Steel II:
- When installing the game from the steam launcher, it tries to install MS .NET 4.0 (for cold steel I) / .NET 4.7 (for cold steel II). This break the game and make them unable to launch. In fact you get the same error as reported here : https://github.com/ValveSoftware/Proton/issues/1048 . To launch the game, you have to use wine-mono
- The intro video is not displayed (the sound plays but the video is a black screen). This is a case of a missing video codec. Installing k-lite codec pack MEGA in the prefix and disabling winegstreamer fix the issue for me.

The following issues apply for The Legend of Heroes: Trails of Cold Steel II only:
- It need protontricks 748490 quartz to start without crashing
- When starting a new game, the player is asked to load a save from the previous game (The Legend of Heroes: Trails of Cold Steel). The game cannot find the save as the save path is not shared between prefixes. The location of the game saves are inside steamapps/compatdata/748490/pfx/drive_c/users/steamuser/Saved Games/Falcom/... .This could be fixed by making the 'Saved Games' folder a symlink to a folder inside the home folder in the same way as wine does for desktop/my documents/my music/etc.

Once these issues have been addressed, it will be possible to mark this series as platinum on protondb as both games work perfectly from the start to the end if they manage to start.

## Reproduction
Try to launch the game after downloading it from the steam app.

**Comment by kisak-valve** [comment]
The Legend of Heroes: Trails of Cold Steel II (748490)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2677.
@goldhawlk posted on 2019-05-12T13:55:54:

# Whitelist Request
- Name of the game to be whitelisted: The Legend of Heroes: Trails of Cold Steel II
- Steam AppID of the game: 748490

## System Information
- GPU: RX580
- Driver/LLVM version: 4.5.13536 Compatibility Profile Context 18.30.2.15
- Distro version: Ubuntu 16.04.6
- Link to full system information report as [Gist]: https://gist.github.com/goldhawlk/c2e9dffe0d289dfa89bbfbf788b288d2
- Proton version: 4.2-3

## I confirm:
- [x] that pressing the `Play` button in the Steam client is sufficient.
- [ ] that [runtime config options](https://github.com/ValveSoftware/Proton#runtime-config-options) are necessary to run the game.
- [ ] that no workarounds other than PROTON_* env variables are necessary.

## Issues
- [x] I haven't experienced any issues.
- [x] There are no issues left open for this game.
- [ ] Although I consider the gaming experience equal to Windows there are remaining issues:

**Comment by kisak-valve** [comment]
Hello @enimog, can you check if this game (Cold Steel II) has any issues with Proton 4.2-3 that would block it from being whitelisted?

**Comment by enimog** [comment]
Tested with a clean prefix (I deleted the old one) and Proton 4.2-3, here are the result I get:
- The configuration tool that comes with the game start properly and I can edit settings no problem.
- With the default configuration, I see some wierd glitch when the opening video start, might be an issue with my system or maybe not... VLC is saying the codec used for the intro video is "DivX MPEG-4 Video (DX50)". Is that supported by proton right now? the sound is playing in the background properly and if I hit any key the game skip the intro video and goes to the main menu. From there, I am able to load one of my saves and play the game normally. By normally I mean that everything is working as expected. Using the configuration tool I can disable all videos from playing and this way I don't see any wierd stuff.

**Comment by RAMChYLD** [comment]
- GPU: AMD Radeon Vega 56
- Driver/LLVM version: `21.3~git2108310600.213548~oibaf~f/12.0.1-1~oibaf~f`
- Kernel version: Linux 5.11.0-27-lowlatency
- Link to full system information report as Gist: https://gist.github.com/RAMChYLD/8bd21ed8b8387d53134beba9c2e8912e
- Proton version: Experimental

With the latest version of proton experimental video is not playing and installing xvid does not resolve the issue anymore. Now it shows a test pattern with a counter and static.

**Comment by zany130** [comment]
unplayable on recent versions of proton constantly crashes and audio randomly breaks  after a few hours of play ( audio stops playing then the game crashes when it goes to load a new scene)
[steam-Trails of Cold Steel II.log](https://github.com/ValveSoftware/Proton/files/7811448/steam-Trails.of.Cold.Steel.II.log)

**Comment by zany130** [comment]
still getting this issue thought I would post a newer proton [log](https://drive.google.com/file/d/1UyFURbV3TOVB8O5N7lfU2Yd1VgmIbcxh/view?usp=sharing) and sys [info](https://gist.github.com/zany130/6eb9b76c72b5ba8743c7c94386839575) since I forgot to for my previous post

**Comment by zany130** [comment]
if anyone runs into this issue on Linux I fixed it by disabling the audio patches in senpatcher. Game works flawlessly (including videos) now on proton 7.0-1

**Comment by rvalles** [comment]
> if anyone runs into this issue on Linux I fixed it by disabling the audio patches in senpatcher. 

That's the first thing I tried. No dice.

Audio issue will happen a lot or not at all depending on the area, which is what I suspect you're experiencing.

**Comment by zany130** [comment]
yeah may have prematurely called it  fixed by disabling the audio patches in senpatcher  got it to happen again with no patches.
seems really busy area's with a lot of different sound sources ( like in front of barahard cathedral where there sfx from the water fountain and the npc footsteps from walking around) is more likely to trigger the bug. and causes `warn:seh:OutputDebugStringA "Command buffer maxed out so refused execution of command: 63504 / 65536 bytes\n"` to appear in the proton log at the same time the audio loss happens which AdmiralCurtiss from senpatcher mentioned might be related

**Comment by AdmiralCurtiss** [comment]
To clarify, that message is from CS2 itself, not Proton or XAudio2. As far as I understand from reverse-engineering anyway, the way the audio logic in CS2 works is that there's a command ringbuffer that the main thread pushes commands into (play SFX, play voice, switch BGM, etc.), and a separate audio thread then processes these into actual XAudio2 calls.

The log message is repeated a lot after the bug triggers, with the same amount of bytes every time, which leads me to believe that the audio thread either dies or deadlocks on Proton and thus no new commands are processed and removed from the buffer anymore, and the main thread stops pushing new ones because the ringbuffer is full, printing the message in the process.
